### DISCUSSION TITLE
3.2 on Ubuntu 24.04: Polkit dialog not working?

### DISCUSSION THREAD

**Discussion by shaine** [regolith-linux/regolith-desktop#1030] [Q&A]
Hello!

I have Regolith 3.2 beta 2 installed on Ubuntu 24.04. I've noticed that I cannot install Steam while in i3, it seems to have trouble authenticating for some reason. I was able to install it when logged into Unity though. 

To recreate, open App Center, search Steam, click Install. In i3, it displays an authentication error dialog box. In Unity, the screen blurs and an authentication box pops up to request a password to escalate privileges.

I don't have the exact logs now, but when trying to install directly from the .deb file, a terminal-esque UI came up to ask for privilege escalation, and after providing the correct password, it output some error about being unable to communicate with the polkit daemon. I verified that the polkit daemon was running at that time. To try to recreate this:

wget https://cdn.akamai.steamstatic.com/client/installer/steam.deb
sudo dpkg -i steam.deb

and then try to launch Steam. It should ask for a password, if its dependencies are not already available on the system, and then fail. (I can't recreate at this exact moment because I now have the deps and don't know how to purge them to try again.)

**Comment by shaine** [comment]
![Screenshot from 2024-07-09 11-59-08](https://github.com/regolith-linux/regolith-desktop/assets/204734/a260996f-a016-461f-8183-603ee5b50246)

**Comment by ibejohn818** [comment]
Here's what I do on ubuntu 23.10 when needing to authenticate via policy-kit...
In a terminal window, run the following to start the policy-kit agent    

``` shell
 sudo /usr/lib/policykit-1-gnome/polkit-gnome-authentication-agent-1
```
With the agent running, you should be able to authenticate VIA the GUI

After authenticating you can stop the agent ... for a more perm solution, you can start the agent in your profile when you login and send it to the background.

**Reply by shaine** [reply]
`policykit-1-gnome` has been replaced in Ubuntu 24.04 by something I'm struggling to learn about and is no longer installed by default. That package is still installable and your answer works after installing it. Decent workaround for now, thank you!

**Comment by kgilmer** [comment]
I'm guessing this is related to `xdg-desktop-portal` and specifically the WIP implementation supporting Regolith.  @SoumyaRanjanPatnaik do you know if this behavior is expected w/ our current implementation?

**Comment by shaine** [comment]
As another data point, 1password has the option to unlock using system authentication. This feature automatically fails when it attempts to trigger.  ibejohn818's workaround resolves it, though I'd prefer to figure out how to use whatever it is that Unity is tapping into, since I assume `policykit-1-gnome` is now a legacy package given its exclusion from 24.04.
